web/api/v1/api.go

```diff
@@ -81,6 +82,33 @@ const (
 	errorNotFound    errorType = "not_found"
 )
 
+// partialResponseStrategy controls the behaviour of given server API on how to treat temporary unavailability of remote components.
+// For example configured remote read APIs in case of Prometheus.
+// TODO(bwplotka): Move to storage and pass through hints so we can abort early if chosen.
+type partialResponseStrategy string
+
+const (
+	// partialResponseStrategyWarn strategy tells server that inconsistent results given unavailability of remote components are
+	// acceptable as long as warnings are provided in response.
+	partialResponseStrategyWarn = "warn"
+	// partialResponseStrategyAbort strategy tells server that inconsistent results given unavailability of remote components are
```

Comment:
Similarly inconsistent results are never okay. You're conflating a partial data with inconsistent data. I'd also not mention anything about remote data - this could also apply to the local tsdb at some point.